catalog/transformer/catalog-transformer-xml/src/main/java/ddf/catalog/transformer/xml/adapter/AdaptedMetacard.java

```diff
@@ -177,14 +179,12 @@ public final Attribute getAttribute(String name) {
     if (Metacard.ID.equals(name)) {
       return this.id;
     }
-    for (Attribute attribute : attributes) {
-      if (attribute == null || StringUtils.isEmpty(attribute.getName())) {
-        continue;
-      } else if (name.equals(attribute.getName())) {
-        return attribute;
-      }
-    }
-    return null;
+    return attributes
+        .stream()
+        .filter(attr -> !StringUtils.isEmpty(attr.getName()))
```

Comment:
❗️ Need to filter for null entries in the `attributes` set: `.filter(Objects::nonNull)`.

Reply:
Actually, I don't think it's possible to get `null` attributes since `setAttribute` does the check. Are there other avenues by which this can occur?

Reply:
I would almost 100% agree, as this method and `setAttribute` are both final and `setAttribute` protects against nulls being inserted; however, the protected `getAttributes` method leaks the collection and could allow modifications to it. If we were to guard against that - which we should - with a defensive copy (always the safest, but more expensive) or by decorating it as Unmodifiable, then I would agree here.

Reply:
Ignore. I see you're making a defensive copy after all.